src/service/RemoteNodeService.ts

```diff
@@ -113,4 +123,105 @@ export class RemoteNodeService {
             .filter((i) => i)
             .map((i) => i as RepositoryInfo);
     }
+
+    public async getRestUrls(presetData: ConfigPreset): Promise<string[]> {
```

Comment:
There are multiple methods taking `presetData` as an argument. Since all of the methods are instance methods, might be better to remove it from method arguments and move it to the constructor instead and access it with `this.presetData`.